test/tools/libtesteth/FillJsonFunctions.cpp

```diff
@@ -0,0 +1,228 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file
+ * Helper functions to work with json::spirit and test files
+ */
+#include <libethereum/Transaction.h>
+#include <libdevcore/CommonData.h>
+
+#include <test/tools/libtesteth/TestHelper.h>
+#include <test/tools/libtesteth/TestOutputHelper.h>
+#include <test/tools/libtesteth/Options.h>
+#include <test/tools/libtesteth/ImportTest.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::test;
+
+namespace dev
+{
+namespace test
+{
+
+
+json_spirit::mObject fillJsonWithTransaction(Transaction const& _txn)
+{
+	json_spirit::mObject txObject;
+	txObject["nonce"] = toCompactHex(_txn.nonce(), HexPrefix::Add, 1);
+	txObject["data"] = _txn.data().size() ? toHex(_txn.data(), 2, HexPrefix::Add) : "";
+	txObject["gasLimit"] = toCompactHex(_txn.gas(), HexPrefix::Add, 1);
+	txObject["gasPrice"] = toCompactHex(_txn.gasPrice(), HexPrefix::Add, 1);
+	txObject["r"] = toCompactHex(_txn.signature().r, HexPrefix::Add, 1);
+	txObject["s"] = toCompactHex(_txn.signature().s, HexPrefix::Add, 1);
+	txObject["v"] = toCompactHex(_txn.signature().v + 27, HexPrefix::Add, 1);
+	txObject["to"] = _txn.isCreation() ? "" : "0x" + toString(_txn.receiveAddress());
+	txObject["value"] = toCompactHex(_txn.value(), HexPrefix::Add, 1);
+	ImportTest::makeAllFieldsHex(txObject);
+	return txObject;
+}
+
+json_spirit::mObject fillJsonWithStateChange(State const& _stateOrig, eth::State const& _statePost, std::vector<dev::eth::detail::Change> const& _changeLog)
+{
+	json_spirit::mObject oState;
+	if (!_changeLog.size())
+		return oState;
+
+	typedef dev::eth::detail::Change::Kind changeKind;
+	typedef dev::eth::detail::Change changeStruct;
+	vector<changeStruct> changeLog = _changeLog;
+
+	//Sort the vector by address field
+	std::sort(changeLog.begin(), changeLog.end(),
+		[ ]( const changeStruct& lhs, const changeStruct& rhs )
+		{
+			return lhs.address < rhs.address;
+		}
+	);
+
+	std::ostringstream log;
+	json_spirit::mObject o;
+
+	json_spirit::mArray agregatedBalance;
+	json_spirit::mArray agregatedStorage;
+	changeStruct popChange = changeLog.at(0);
+	std::map<string, ostringstream> logInfo;
+	u256 tmpBalance = _stateOrig.balance(popChange.address);
+	for (size_t i = 0; i < changeLog.size(); i++)
```

Comment:
This loop probably works correctly, but I think expressing it something like this would be clearer:
```
size_t i = 0;
while (i < changeLog.size())
{
	Addres currentAddress = changeLog[i].address;

	string before;
	string after;
	json_spirit::mArray record;
	json_spirit::mArray agregatedBalance;
	json_spirit::mArray agregatedStorage;
	std::map<string, ostringstream> logInfo;

	while (i < changeLog.size() && changeLog[i].address == currentAddress)
	{
		// handle changeLog[i]

		++i;
	}
}
```